archive/100-theorems-list/70_perfect_numbers.lean

```diff
@@ -0,0 +1,671 @@
+import data.nat.totient
+import number_theory.quadratic_reciprocity
+
+open_locale classical
+open_locale big_operators
+
+section definitions
+
+variable (n : ℕ)
+
+def divisors : finset ℕ :=
+finset.filter (λ x : ℕ, x ∣ n) (finset.Ico 1 (n + 1))
+
+def proper_divisors : finset ℕ :=
+finset.filter (λ x : ℕ, x ∣ n) (finset.Ico 1 n)
+
+lemma not_proper_self :
+¬ n ∈ proper_divisors n :=
+begin 
+  rw proper_divisors,
+  simp,
+end
+
+lemma mem_divisors {m : ℕ} : 
+n ∈ divisors m ↔ n ∣ m ∧ m ≠ 0:=
+begin
+  rw divisors,
+  simp,
```

Comment:
For reasons of maintainability, we avoid "non-terminal `simp`". You can convert this to a `simp only ...` by first replacing it with `squeeze_simp` and looking at the trace message.

Reply:
Gotcha. This one's replaced, and I think I remembered to squeeze_simp everywhere else.